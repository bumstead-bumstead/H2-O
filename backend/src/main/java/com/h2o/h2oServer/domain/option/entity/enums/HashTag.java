package com.h2o.h2oServer.domain.option.entity.enums;

public enum HashTag {
    LEISURE("레저"),
    SPORTS("스포츠"),
    MOVIE("영화감상"),
    CAMPING("캠핑"),
    STYLE("스타일"),
    SMART("스마트"),
    COMFORTABLE("쾌적"),
    LONG_DISTANCE_DRIVING("장거리 운전"),
    SHORT_DISTANCE_DRIVING("단거리 운전"),
    CHILD_COMMUTE("자녀 통학"),
    COMMUTE("출퇴근"),
    BUSINESS_TRIP("출장"),
    SAFETY("안전"),
    PARKING("주차/출차"),
    CHILDREN("자녀"),
    SINGLE_HOUSEHOLD("1인가구"),
    COUPLE("부부"),
    MULTI_PERSON_FAMILY("다인가족"),
    EYE_CATCHER("눈길"),
    CLOUDY_DAY("흐린날"),
    BEGINNER_DRIVING("초보운전"),
    HIGHWAY("고속도로"),
    NATIONAL_ROAD("국도"),
    NIGHT("야간"),
    URBAN("도시"),
    RURAL("시외"),
    FEMALE("여성"),
    MALE("남성"),
    PET("반려견/묘");

    private final String label;

    HashTag(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }

    public static HashTag fromLabel(String label) {
        for (HashTag hashTag : HashTag.values()) {
            if (hashTag.getLabel().equals(label)) {
                return hashTag;
            }
        }
        throw new IllegalArgumentException("존재하지 않는 해시태그 라벨 : " + label);
    }
}
