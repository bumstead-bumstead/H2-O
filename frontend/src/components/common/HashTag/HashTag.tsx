import styled from '@emotion/styled';

const HashTag = styled.div(
  {
    display: 'flex',
    alignItems: 'center',
    padding: '2px 6px',
    borderRadius: '2px',
  },
  ({ theme }) => ({
    ...theme.typography.TextKRRegular12,
    backgroundColor: theme.colors.hashTagBg,
    color: theme.colors.gray50,
  }),
);

export default HashTag;
