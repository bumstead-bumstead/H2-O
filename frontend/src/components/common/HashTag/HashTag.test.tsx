import HashTag from './HashTag';
import { render, screen } from '@/tests/test-util';

describe('HashTag 컴포넌트 테스트', () => {
  it('HashTag가 정상적으로 렌더링된다.', () => {
    render(<HashTag>해시태그</HashTag>);

    expect(screen.getByText('해시태그')).toBeInTheDocument();
  });

  it('HashTag 스냅샷 테스트', () => {
    render(<HashTag>해시태그</HashTag>);

    expect(screen.getByText('해시태그')).toMatchSnapshot();
  });
});
