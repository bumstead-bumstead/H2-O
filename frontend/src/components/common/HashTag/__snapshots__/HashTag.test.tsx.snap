// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HashTag 컴포넌트 테스트 > HashTag 스냅샷 테스트 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2px 6px;
  border-radius: 2px;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 18px;
  letter-spacing: -0.03em;
  background-color: #75757580;
  color: #F7F8F9;
}

<div
  class="emotion-0"
>
  해시태그
</div>
`;
